reframe/core/pipeline.py

```diff
@@ -1240,17 +1240,12 @@ def compile(self):
             self.build_system.executable = self.executable
 
         if isinstance(self.build_system, EasyBuild):
-            if not self.build_system.installpath:
-                self.build_system.installpath = os.path.join(self._stagedir,
-                                                             'rfm_easybuild')
-            if not self.build_system.buildpath:
-                self.build_system.buildpath = os.path.join(self._stagedir,
-                                                           'rfm_easybuild',
-                                                           'build')
-
-            if not self.build_system.sourcepath:
-                self.build_system.sourcepath = os.path.join(self._stagedir,
-                                                            'rfm_easybuild')
+            self.build_system._eb_sandbox = os.path.join(self._stagedir,
+                                                         'rfm_easybuild')
```

Comment:
The `RegressionTest` should not care about this directory. It should the responsibility of the build system. And I don't think that the `stagedir` is needed also here. The compilation is always invoked from the stage directory.